src/openforms/submissions/pricing.py

```diff
@@ -21,16 +21,8 @@ def get_submission_price(submission: "Submission") -> Decimal | None:
 
     :param submission: the :class:`openforms.submissions.models.Submission: instance
       to calculate the price for.
-    :return: the calculated price, or ``None`` if no payment is required.
+    :return: the calculated price.
```

Comment:
below `assert submission.form.product`, I'd add another:

```py
assert submission.form.product.price, "'get_submission_price' may only be called for forms that require payment"
```
That way, we can make the intent clear after removing the guard clause.